interpreter/cling/lib/Interpreter/DynamicLibraryManager.cpp

```diff
@@ -205,6 +206,18 @@ namespace cling {
 
     std::string errMsg;
     DyLibHandle dyLibHandle = platform::DLOpen(canonicalLoadedLib, &errMsg);
+    llvm::StringRef EM(errMsg);
+    // If error message contains "undefined symbol", it means that we failed to load dependency
+    // libraries. We emit callback to LazyFunctionCallback with undefined symbol and try to load
+    // the library which has undef's definition.
+    if (EM.contains("undefined symbol: ")) {
+      std::pair<llvm::StringRef, llvm::StringRef> hoge = EM.split("undefined symbol: ");
+      std::string undef = std::string(hoge.second);
+      if (InterpreterCallbacks* C = getCallbacks())
+        C->LazyFunctionCallback(undef);
+      return loadLibrary(libStem, permanent, resolved);
```

Comment:
This should go in the callback in TCling.

Reply:
No this is for dlopening the same library (not the dependency one) because undefined symbol might not be just one

Reply:
Ok, that was not obvious reading the diff on my phone. Could you add a comment saying: 'Retry opening the library because the callback may have provided the undefined symbol.'

Also, shouldn't we call `loadLibrary` only when there are callbacks and the result of the callback was 'success'?